pandas/sparse/tests/test_sparse.py

```diff
@@ -770,6 +770,32 @@ def test_combine_first(self):
         assert_sp_series_equal(result, result2)
         assert_sp_series_equal(result, expected)
 
+class TestSparseHandlingMultiIndexes(tm.TestCase):
+
+    def setUp(self):
+        miindex = pd.MultiIndex.from_product([["x","y"], ["10","20"]],names=['row-foo', 'row-bar'])
+        micol = pd.MultiIndex.from_product([['a','b','c'], ["1","2"]],names=['col-foo', 'col-bar'])
+        dense_multiindex_frame = pd.DataFrame(index=miindex, columns=micol).sortlevel().sortlevel(axis=1)
+        self.dense_multiindex_frame = dense_multiindex_frame.fillna(value=3.14)
+
+    def test_to_sparse_preserve_multiindex_names_on_columns(self):
+        sparse_multiindex_frame = self.dense_multiindex_frame.to_sparse()
+        self.assertTrue(self.dense_multiindex_frame.columns.equals(sparse_multiindex_frame.columns))
```

Comment:
Digging into assert_sp_frame_equal it appears that if I feed it a dense reference frame and a sparse frame it converts the sparse frame to dense and compares them in the dense state. This would make an assert_sp_frame of the sparse and dense matrices the same as the round trip test (i.e. it only tests the sparse frame after passing it through to_dense).

I am trying to actually test the column names while still sparse against the reference column names to make sure we don't wind up just preserving the column names during conversion and putting them back on when going back to_dense. So I actually want a test of the sparse column names without making it dense again. I agree the round_trip_test goes much better with assert_frame_equal, but it looks like testing the column names are actually valid while sparse can't use assert_sp_frame_equal?